src/legacy/trulens_eval/__init__.py

```
@@ -0,0 +1,100 @@
+
+with imports_utils.OptionalImports(
+    messages=imports_utils.REQUIREMENT_PROVIDER_LITELLM
+):
+    from trulens.providers.litellm.provider import LiteLLM
+
+with imports_utils.OptionalImports(
+    messages=imports_utils.REQUIREMENT_PROVIDER_BEDROCK
+):
+    from trulens.providers.bedrock.provider import Bedrock
+
+with imports_utils.OptionalImports(
+    messages=imports_utils.REQUIREMENT_PROVIDER_OPENAI
+):
+    from trulens.providers.openai.provider import AzureOpenAI
+    from trulens.providers.openai.provider import OpenAI
+
+with imports_utils.OptionalImports(
+    messages=imports_utils.REQUIREMENT_PROVIDER_CORTEX
+):
+    from trulens.providers.cortex.provider import Cortex
+
+with imports_utils.OptionalImports(
+    messages=imports_utils.REQUIREMENT_PROVIDER_HUGGINGFACE
+):
+    from trulens.providers.huggingface.provider import Huggingface
+    from trulens.providers.huggingface.provider import HuggingfaceLocal
+
+with imports_utils.OptionalImports(
+    messages=imports_utils.REQUIREMENT_PROVIDER_LANGCHAIN
+):
+    from trulens.providers.langchain.provider import Langchain
+
+with imports_utils.OptionalImports(
+    messages=imports_utils.REQUIREMENT_INSTRUMENT_LANGCHAIN
+):
+    from trulens.instrument.langchain.tru_chain import TruChain
+
+with imports_utils.OptionalImports(
+    messages=imports_utils.REQUIREMENT_INSTRUMENT_LLAMA
+):
+    from trulens.instrument.llamaindex.tru_llama import TruLlama
+
+with imports_utils.OptionalImports(
+    messages=imports_utils.REQUIREMENT_INSTRUMENT_NEMO
+):
+    from trulens.instrument.nemo.tru_rails import TruRails
+
+__all__ = [
+    "Tru",  # main interface
+    # app types
+    "TruBasicApp",
+    "TruCustomApp",
+    "TruChain",
+    "TruLlama",
+    "TruVirtual",
+    "TruRails",
+    # app setup
+    "FeedbackMode",
+    # feedback setup
+    "Feedback",
+    "Select",
+    # feedback providers
+    "Provider",
+    "AzureOpenAI",
+    "OpenAI",
+    "Langchain",
+    "LiteLLM",
```

What would happen to this if those imports like `Bedrock` is not present, would it give a silent error? or do we create this list based on the conditional imports?

---

Same as before, the optional imports mechanism will make Bedrock a dummy class.